h2o-core/src/main/java/water/util/IcedHashMapGeneric.java

```diff
@@ -0,0 +1,192 @@
+package water.util;
+
+import water.AutoBuffer;
+import water.Freezable;
+import water.H2O;
+import water.Iced;
+import water.nbhm.NonBlockingHashMap;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Iced / Freezable NonBlockingHashMap abstract base class.
```

Comment:
Can you add some comments on why this was implemented, and fix the above comment - this isn't an abstract base class.